internal/telemetry/appinsights.go

```diff
@@ -0,0 +1,67 @@
+package telemetry
```

Comment:
>ackage telemetry [](http://example.com/codeflow?start=1&length=16)

it could be useful to define a method that accepts event name and properties plus a func() as input. It would run the function with a timer and add an elapsed time property to the telemetry event before posting it.

Something like
```
TrackTimedOperation(name string, properties map[string]string, func ())
```
then we can measure how long it takes to deploy things without trying to look for a "begin" event and its corresponding "end" event.

Reply:
This is a good suggestion , but i would like to add one POV here . Instead of every granular event to have start and end time , we should have these func parmater with specified events (such as deploy and create).

Reply:
Yes I don't think every operation needs an elapsed time property, just certain ones. At the lowest level, though, the function just needs to take an event name and properties. The application-level code will decide which events need to use it.